compiler/rustc_session/src/filesearch.rs

```diff
@@ -193,6 +209,18 @@ pub fn get_or_default_sysroot() -> Result<PathBuf, String> {
         }
     }
 
+    // Select a default path based on the platform this Rust binary is currently running on
+    fn default_from_current_platform() -> Result<PathBuf, String> {
+        #[cfg(target_os = "linux")]
+        {
+            if let Ok(p) = default_from_current_exe() {
+                return Ok(p);
+            }
+        }
+
+        default_from_rustc_driver_dll()
+    }
+
```

Comment:
for linux, this will bring back https://github.com/rust-lang/rust/issues/98832 problem again.